pdc/apps/package/serializers.py

```diff
@@ -221,3 +222,79 @@ class BuildImageRTTTestsSerializer(StrictSerializerMixin, serializers.ModelSeria
     class Meta:
         model = models.BuildImage
         fields = ('id', 'build_nvr', 'format', 'test_result')
+
+
+class RepositoryField(serializers.SlugRelatedField):
+    def __init__(self, **kwargs):
+        super(RepositoryField, self).__init__(slug_field='id', queryset=Repo.objects.all(), **kwargs)
+
+    def to_representation(self, value):
+        export_data = value.export()
+        export_data["id"] = value.id
+        return export_data
+
+
+class ReleasedFilesSerializer(StrictSerializerMixin, serializers.ModelSerializer):
+    # base on content.format, currently rpm.srpm_name, rpm.version, rpm.release
+    build = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.srpm_name
+    package = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.filename
+    file = serializers.CharField(required=False, default=None)
+    # base on content.format, pk
+    file_primary_key = serializers.IntegerField(allow_null=False)
+    repo = RepositoryField()
+    released_date = serializers.DateField(required=False)
+    release_date = serializers.DateField()
+    created_at = serializers.DateTimeField(required=False, read_only=True)
+    updated_at = serializers.DateTimeField(required=False, read_only=True)
+    zero_day_release = serializers.BooleanField(required=False, default=False)
+    obsolete = serializers.BooleanField(required=False, default=False)
+
+    class Meta:
+        model = models.ReleasedFiles
+        fields = ('id', 'build', 'package', 'file', 'file_primary_key', 'repo',
+                  'released_date', 'release_date', 'created_at', 'updated_at',
+                  'zero_day_release', 'obsolete')
+
+    def validate(self, data):
+        if "repo" in data:
+            repo_format = data["repo"].content_format
+            repo_name = data["repo"].name
+            if str(repo_format) != "rpm":
+                raise serializers.ValidationError(
+                    {'detail': 'Currently we just support rpm type of repo, the type of %s is %s ' % (repo_name, repo_format)})
+
+        if "file_primary_key" in data:
+            d = models.RPM.objects.get(id=data["file_primary_key"])
+            if "build" in data:
+                if data["build"]:
```

Comment:
Value of "build" field can be empty?

Reply:
Yes, because if in the future there is no field called build in other type of packages.
Currently just use for rpm.